SimplenoteUITests/NoteList.swift

```diff
@@ -39,7 +39,9 @@ class NoteList {
         guard !NoteList.isAllNotesListOpen() else { return }
         print(">>> Opening \"All Notes\"")
         Sidebar.open()
-        app.tables.staticTexts[UID.Button.allNotes].tap()
+        let allNotesButton = app.tables.staticTexts[UID.Button.allNotes]
+        guard allNotesButton.waitForExistence(timeout: averageLoadTimeout) else { return }
```

Comment:
That's a good question... Failing it makes sense, because the execution will go wrong for sure if it's not found, so we will save time before reaching the next inevitable error/fail. On the other hand, the current approach in the project is having assertions only inside asserting methods or tests, and not having them in methods that 'do stuff'. What is your opintion about this?

Reply:
> On the other hand, the current approach in the project is having assertions only inside asserting methods or tests, and not having them in methods that 'do stuff'.

Good point. Let's stick with that for the moment and have a dedicated experimentation round for (possibly) changing the approach in the future.

I think the sooner we fail the better. That's particularly true for UI tests, which are notoriously time consuming.

Reply:
This is a short WWDC session on the topic of handling failures in tests: https://developer.apple.com/videos/play/wwdc2020/10091/

There may be something useful for us in there. Just because Apple does something it doesn't mean it's right, still, it may be good food for thought. (_I could be wrong, though, I remember liking it, but I don't really remember the exact recommendations. I should re-watch it..._)